homeassistant/components/flux/config_flow.py

```diff
@@ -0,0 +1,268 @@
+    MODE_RGB,
+    MODE_XY,
+)
+from .switch import CONF_DISABLE_BRIGHTNESS_ADJUST
+
+_LOGGER = logging.getLogger(__name__)
+
+MINIMAL_FLUX_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_LIGHTS): EntitySelector(
+            EntitySelectorConfig(domain=Platform.LIGHT, multiple=True)
+        ),
+    }
+)
+
+allowed_colortemp_range = ColorTempSelectorConfig(
+    {
+        "min_mireds": color_temperature_kelvin_to_mired(40000),
+        "max_mireds": color_temperature_kelvin_to_mired(1000),
+    }
+)
+
+
+def default_settings():
+    """Return object with the default settings for the Flux integration."""
+    settings_dict = {}
+    settings_dict[CONF_START_CT] = DEFAULT_START_COLOR_TEMP_KELVIN
+    settings_dict[CONF_SUNSET_CT] = DEFAULT_SUNSET_COLOR_TEMP_KELVIN
+    settings_dict[CONF_STOP_CT] = DEFAULT_STOP_COLOR_TEMP_KELVIN
+    settings_dict[CONF_ADJUST_BRIGHTNESS] = True
+    settings_dict[CONF_MODE] = DEFAULT_MODE
+    settings_dict[CONF_INTERVAL] = {"seconds": 30}
+    settings_dict[ATTR_TRANSITION] = {"seconds": 30}
+    return settings_dict
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Flux."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry) -> OptionsFlow:
+        """Get the options flow for the Flux component."""
+        return OptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle the initial step."""
+
+        if user_input is not None:
+            user_input.update(default_settings())
+            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=MINIMAL_FLUX_SCHEMA,
+        )
+
+    async def async_step_import(self, yaml_config: ConfigType) -> FlowResult:
+        """Handle import from configuration.yaml."""
+        # start with the same default settings as in the UI
+        entry_data = default_settings()
+
+        # remove the old two very similar options
+        brightness = yaml_config.pop(CONF_BRIGHTNESS, False)
+        disable_brightness_adjust = yaml_config.pop(
+            CONF_DISABLE_BRIGHTNESS_ADJUST, False
+        )
+
+        # combine them into the "new" option
+        if brightness or disable_brightness_adjust:
+            entry_data[CONF_ADJUST_BRIGHTNESS] = False
+
+        entry_data[CONF_INTERVAL] = {"seconds": yaml_config.pop(CONF_INTERVAL)}
+        entry_data[ATTR_TRANSITION] = {"seconds": yaml_config.pop(ATTR_TRANSITION)}
+
+        if CONF_START_TIME in yaml_config:
+            entry_data[CONF_START_TIME] = str(yaml_config.pop(CONF_START_TIME))
+
+        if CONF_STOP_TIME in yaml_config:
+            entry_data[CONF_STOP_TIME] = str(yaml_config.pop(CONF_STOP_TIME))
+
+        yaml_config.pop("platform")
```

Comment:
You don't really have to pop everything, you just get the data you need and that's it